### PR TITLE
📚 Various textual/flow improvements

### PR DESCRIPTION
**Pull request by ltalirz**
* developers: link to "customise" section instead of repeating content
 * maintainers: split off release docs into separate file

Various textual/flow improvements

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -97,4 +95,5 @@ developers/roles.md
 :caption: Maintainers
 
 maintainers/develop.md
+maintainers/release.md
```
Comment (chrisjsewell) [inline docs/index.md:98]: did you forget to add this to the commit?

Comment (ltalirz) [reply docs/index.md:98]: yep, fixed!

File: docs/maintainers/release.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+* see [cloud build instructions](../developers/build-cloud.md).
+
+:::{note}
+Here are some indicative prices for hosting public images and disk pricing:
+
+* Image pricing (for us to host it):
+  * magnetic: 8.5c/GB/month (GCP)
+* Disk pricing (for users):
+  * magnetic: 4c/GB/month (GCP), 2.5c/GB/month (AWS)
+  * SSD: 10c/GB/month (AWS)
```
Comment (chrisjsewell) [inline docs/maintainers/release.md:80]:
```suggestion
  * SSD: 10c/GB/month (AWS)
:::

```

**Review by chrisjsewell — APPROVED** [review]
just one syntax nitpick, otherwise looks good 👍

**Pull request merged**